public/Read-DbaXEFile.ps1

```diff
@@ -54,83 +54,80 @@ function Read-DbaXEFile {
-                    return
-                }
-
-                if ($file.TargetFile.Length -eq 0) {
-                    Stop-Function -Message "This session does not have an associated Target File."
-                    return
+            if ($pathObject -is [System.String]) {
+                $files = $pathObject
+            } elseif ($pathObject -is [System.IO.FileInfo]) {
+                $files = $pathObject.FullName
+            } elseif ($pathObject -is [Microsoft.SqlServer.Management.XEvent.Session]) {
+                if ($pathObject.TargetFile.Length -eq 0) {
+                    Stop-Function -Message "The session [$pathObject] does not have an associated Target File." -Continue
                 }
 
-                $instance = [dbainstance]$file.ComputerName
-
+                $instance = [dbainstance]$pathObject.ComputerName
                 if ($instance.IsLocalHost) {
-                    $currentfile = $file.TargetFile
+                    $targetFile = $pathObject.TargetFile
                 } else {
-                    $currentfile = $file.RemoteTargetFile
+                    $targetFile = $pathObject.RemoteTargetFile
                 }
+
+                $targetFile = $targetFile.Replace('.xel', '*.xel').Replace('.xem', '*.xem')
+                $files = Get-ChildItem -Path $targetFile | Where-Object Length -gt 0
+                Write-Message -Level Verbose -Message "Received $($files.Count) files based on [$targetFile]"
+            } else {
+                Stop-Function -Message "The Path [$pathObject] has an unsupported file type of [$($pathObject.GetType().FullName)]."
             }
 
-            $accessible = Test-Path -Path $currentfile
-            $whoami = whoami
+            foreach ($file in $files) {
+                $accessible = Test-Path -Path $file
+                $whoami = whoami
 
-            if (-not $accessible) {
-                if ($file.Status -eq "Stopped") { continue }
-                Stop-Function -Continue -Message "$currentfile cannot be accessed from $($env:COMPUTERNAME). Does $whoami have access?"
-            }
+                if (-not $accessible) {
+                    if ($pathObject.Status -eq "Stopped") { continue }
+                    Stop-Function -Continue -Message "$file cannot be accessed from $($env:COMPUTERNAME). Does $whoami have access?"
+                }
 
-            if ($Raw) {
-                return (SqlServer.XEvent\Read-SqlXEvent -FileName $currentfile)
-            }
+                # use the SqlServer\Read-SqlXEvent cmdlet from Microsoft
+                # because the underlying Class uses Tasks
+                # which is hard to handle in PowerShell
 
-            # use the SqlServer.XEvent\Read-SqlXEvent cmdlet from Microsoft
-            # because the underlying Class uses Tasks
-            # which is hard to handle in PowerShell
+                if ($Raw) {
+                    SqlServer\Read-SqlXEvent -FileName $file
```

Comment:
Not fond that we tell everyone that we know dbatools v2 does not play nice with sqlserver module when they are in the same session and turn around and have it in a command. What affect is had if someone uses this command, sqlserver imports, and then try using commands that do not play well?

This is missing all kinds of validation that (1) the module is installed and (2) error handling on the calling the command. (There is not one try/catch in this command).